product_multicompany_submit/tests/test_multicompany_submit.py

```diff
@@ -0,0 +1,165 @@
+                                "product_multicompany_submit."
+                                "group_multicompany_product_submit"
+                            ),
+                            self.ref("base.group_user"),
+                        ],
+                    )
+                ],
+            }
+        )
+
+        self.user_other = self.env["res.users"].create(
+            {
+                "name": "other user",
+                "login": "other user",
+                "company_ids": self.company_other.ids,
+                "company_id": self.company_other.id,
+                "groups_id": [(6, 0, [self.ref("base.group_user")],)],
+            }
+        )
+
+        self.product = self.env["product.template"].create(
+            {"name": "some product", "company_id": self.company_submitter.id}
+        )
+
+    def test_permissions(self):
+        self.assertEqual(
+            self.product.state_multicompany_submit, "not_submitted"
+        )
+        self.assertEqual(
+            self.product.multicompany_origin_company_id, self.company_submitter
+        )
+        self.assertEqual(self.product.company_id, self.company_submitter)
+
+        # NOT SUBMITTED permissions: approver, other user should not have them
+        with self.assertRaises(AccessError):
+            self.env["product.template"].sudo(self.user_approver.id).browse(
+                self.product.ids
+            ).read(["name"])
+        with self.assertRaises(AccessError):
+            self.env["product.template"].sudo(self.user_other).browse(
+                self.product.ids
+            ).read(["name"])
+
+        # NOT SUBMITTED permissions: submitter should have them
+        self.env["product.template"].sudo(self.user_submitter).browse(
+            self.product.ids
+        ).read(["name"])
+
+        # PENDING permissions: approver should have them
+        self.product.state_multicompany_submit = "pending_approval"
+        self.env["product.template"].sudo(self.user_approver).browse(
+            self.product.ids
+        ).read(["name"])
+
+        # PENDING permissions: other user should still not have them
+        with self.assertRaises(AccessError):
+            self.product.state_multicompany_submit = "pending_approval"
+            self.env["product.template"].sudo(self.user_other).browse(
+                self.product.ids
+            ).read(["name"])
+
+        # APPROVED permissions: everyone
+        self.product.button_multicompany_approve()
+        self.assertEqual(self.product.company_id.ids, [])
+        self.env["product.template"].sudo(self.user_approver).browse(
+            self.product.ids
+        ).read(["name"])
+        self.env["product.template"].sudo(self.user_submitter).browse(
+            self.product.ids
+        ).read(["name"])
+        self.env["product.template"].sudo(self.user_other).browse(
+            self.product.ids
+        ).read(["name"])
+
+    def test_workflow_accept(self):
```

Comment:
For simplifying the test "accept" and "refuse" can you
- create a new class "TestMulticompanyApproval"
- submit the product is the setup
- then only test the action "accept", "refuse"

Also add a test to check that a user that is not in the group approval can not accept or refuse (I think now it's broken)

Reply:
simplified according to your feedback, and now tests permissions for each case